### DISCUSSION TITLE
Usage in NodeJS

### DISCUSSION THREAD

**Discussion by achou11** [protomaps/PMTiles#89] [Q&A] [answered]
Hello! Love all of the work you're doing with PMTiles and it's super exciting to see where it's going. 

From my understanding, the JS library is solely meant for consumption by a client mapping library. Are there plans to be able to work with PMTiles using NodeJS? (if not already supported)

For context, I'm interested in eventually using PMTiles for a tile server that I'm building for my work at Digital Democracy: https://github.com/digidem/mapeo-map-server

It's currently built with Node and just using SQLite as the storage format for the tiles, but I'm sure that there's potential to consider PMTiles :)

**Comment by bdon** [comment, marked as answer]
It should work, because the test suite runs inside NodeJS. But is there a reason the Mapeo map client frontend cannot consume the PMTiles directly instead of having a NodeJS server in the middle, if this is all offline?

**Reply by achou11** [reply]
> It should work, because the test suite runs inside NodeJS.

Ah great! Would be helpful to have that documented

> But is there a reason the Mapeo map client frontend cannot consume the PMTiles directly instead of having a NodeJS server in the middle, if this is all offline?

The goal of the server is to support map management flows that we want to build for common use cases with our partners. On android, it's embedded in the app and accessed via [nodejs-mobile](https://github.com/nodejs-mobile/nodejs-mobile), so offline-first in a different sense 😄 We have plans to use it for our desktop app too (currently built on Electron).

**Reply by bdon** [reply]
FYI, I'm likely going to require Node v18+ going forward, as this is supported in Lambda now and most of the big features (Web streams, isomorphic Fetch) are relevant to PMTiles.
